Master

Alexander66666666 opened the pull request.

Alexander66666666 committed:
Ошибка 500 при запуске докера

Alexander66666666 committed:
Ошибка 500 при запуске докера

VladislavYurin commented on src/main/java/com/example/orderservice/config/SecurityConfig.java:
```diff
@@ -0,0 +1,48 @@
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor
+@SuppressWarnings("unused")
```
почему unused?

VladislavYurin commented on src/main/java/com/example/orderservice/controller/AuthController.java:
```diff
@@ -0,0 +1,57 @@
+        User user = new User();
+        user.setUsername(request.getUsername());
+        user.setPassword(request.getPassword());
+        User savedUser = userService.saveUser(user);
```
я бы сделал так чтобы в метод userService.saveUser передавался сразу AuthRequest request, и уже внутри вся конвертация. Потому что обычно пытаются максимально всю логику вынести из AuthController

VladislavYurin commented on src/main/java/com/example/orderservice/controller/AuthController.java:
```diff
@@ -0,0 +1,57 @@
+    @PostMapping("/register")
+    public ResponseEntity<AuthResponse> register(@Valid @RequestBody AuthRequest request) {
+        if (userService.existsByUsername(request.getUsername())) {
+            return ResponseEntity.badRequest().build();
```
нужно сделать так чтобы нутри userService.existsByUsername при ошибке вылетал кастомный ексепшн. А все кастомные ексепшены надо ловить ControllerAdvice классом, посмотри как это делается, там есть аннотация удобная

VladislavYurin commented on src/main/java/com/example/orderservice/dto/AuthResponse.java:
```diff
@@ -0,0 +1,11 @@
+@AllArgsConstructor
+public class AuthResponse {
+    private String token;
+}
```
нужно реализовать два токена, один accessToken - основной, по нему на все ендпоинты идет проверка, и второй refreshToken (он живет намного дольше - вплоть до 2 недель), и нужен ендпоинт /refresh - туда подается рефреш токен и ответом будет новый аксессТокен и новый (или старый) рефрешТокен. Плюсом можно сюда возвращать роль юзера в системе (админ/юзер)

VladislavYurin commented on src/main/java/com/example/orderservice/dto/OrderRequest.java:
```diff
@@ -0,0 +1,10 @@
+
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+@SuppressWarnings("unused")
```
почему unused?

VladislavYurin commented on src/main/java/com/example/orderservice/dto/OrderResponse.java:
```diff
@@ -0,0 +1,16 @@
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+@SuppressWarnings("unused")
```
почему unused?

VladislavYurin commented on src/main/java/com/example/orderservice/dto/UserResponse.java:
```diff
@@ -0,0 +1,13 @@
+import lombok.Data;
+
+import java.util.UUID;
+@SuppressWarnings("unused")
```
почему unused?

VladislavYurin commented on src/main/java/com/example/orderservice/entity/Order.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.UUID;
+
+@SuppressWarnings("unused")
+@Data
```
Data и Entity в спринге имеют некоторые конфликты и их настоятельно рекомендуется не использовать одновременно

VladislavYurin commented on src/main/java/com/example/orderservice/entity/Status.java:
```diff
@@ -0,0 +1,5 @@
+package com.example.orderservice.entity;
+@SuppressWarnings("unused")
+public enum Status {
```
вынести в папку enum или constant

VladislavYurin commented on src/main/java/com/example/orderservice/entity/User.java:
```diff
@@ -0,0 +1,58 @@
+
+@SuppressWarnings("unused")
+@Data
+@Entity
```
аналогично про unused и data+entity

VladislavYurin commented on src/main/java/com/example/orderservice/service/OrderService.java:
```diff
@@ -0,0 +1,45 @@
+import java.util.UUID;
+@Service
+@RequiredArgsConstructor
+public class OrderService {
```
обычно по солиду делают так:
создают интерфейс OrderService, в нем описание будущих публичных методов, и потом делают OrderServiceImpl который реализует интерфейс OrderService и оверрайдит методы. В других классах поставляется интерфейс.

VladislavYurin commented on src/main/java/com/example/orderservice/service/OrderService.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    public List<Order> getOrdersByUser(User user) {
+        return orderRepository.findByUser(user);
```
а если не найдется?

VladislavYurin commented on src/main/java/com/example/orderservice/service/OrderService.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    public List<Order> getAllOrders() {
+        return orderRepository.findAll();
```
а если не найдется?

VladislavYurin commented on src/main/java/com/example/orderservice/service/OrderService.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    public Optional<Order> getOrderById(UUID id) {
+        return orderRepository.findById(id);
```
а если не найдется?

Alexander66666666 committed:
Исправил, но при запуске приложения, ошибка Caused by: org.flywaydb.core.internal.exception.FlywaySqlException: Unable to obtain connection from database: FATAL: password authentication failed for user "admin"
                                            -------------------------------------------------------------------------------------------------
  SQL State  : 28P01
  Error Code : 0
 Message    : FATAL: password authentication failed for user "admin"
